src/main/java/duke/command/Command.java

```diff
@@ -0,0 +1,198 @@
+package duke.command;
+import duke.dukeException.DukeException;
+import duke.task.*;
+import java.io.*;
+import java.time.LocalDate;
+import java.util.ArrayList;
+
+public class Command {
+    /** The input command by user */
+    public String commandName;
+
+    /**
+     * Class constructor.
+     */
+    public Command(String commandName) {
+        this.commandName = commandName;
+    }
+
+
+    /**
+     * Processes tasks according to the command.
+     *
+     * @param tasks  the tasks before execution.
+     * @throws IOException  If an input or output
+     *                      exception occurred
+     */
+    public void execute(TaskList tasks) throws IOException, DukeException {
+        ArrayList<Task> tList = tasks.tasks;
```

Comment:
Perhaps a plural name tasks would be better? Maybe the name "tList" could be potentially confusing as well when referred to further down or in other locations, as it could mean task list or todo list.